solana/src/burn.rs

```diff
@@ -45,9 +42,16 @@ pub trait SolanaNetwork: Send + Sync + 'static {
         amount: u64,
     ) -> Result<Self::Transaction, Self::Error>;
 
+    async fn sign_transaction(
+        &self,
+        transaction: &Self::Transaction,
+    ) -> Result<Self::Transaction, Self::Error>;
+
     async fn submit_transaction(&self, transaction: &Self::Transaction) -> Result<(), Self::Error>;
 
     async fn confirm_transaction(&self, txn: &Signature) -> Result<bool, Self::Error>;
+
+    async fn check_for_blockhash_not_found_error(&self, err: &Self::Error) -> bool;
```

Comment:
You should have the `Error` type satisfy a separate trait. This is a strange way of doing things, for one because self is completely useless. Also, this function does not need to be async

Reply:
aha, clearly i misunderstood your prior suggestions.   Addressed better here: 5e532bb4f4477c2566991def9436a2680e846d6f